src/calendar-list/index.tsx

```diff
@@ -230,10 +230,8 @@ class CalendarList extends Component<CalendarListProps, State> {
   }
 
   addMonth = (count: number) => {
-    this.updateMonth(this.state.currentMonth.clone().addMonths(count, true));
-  };
-
-  updateMonth(day: XDate) {
+    const day = this.state.currentMonth.clone().addMonths(count, true);
```

Comment:
Something that occur to me that we might have missed. 
There are many ways to clone an object and it's possible the way we clone is expensive especially if we're doing it in many places (which I know we do in Calendars) 

There are many questions about it, for [example](https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript)
And someone mention this benchmark testing -> http://jsben.ch/bWfk9
And it looks like cloning using `object.assign` is best option. 
Maybe we should consider handling that as well (as a different task of course)

Reply:
I agree. We have it alot in Calendars and we should replace it with the best option. I can open a ticket for it